core/network/impl/protocols/sync_protocol_impl.cpp

```diff
@@ -134,6 +136,8 @@ namespace kagome::network {
               host,
               make_protocols(kSyncProtocol, genesis_hash, chain_spec),
               log::createLogger(kSyncProtocolName, "sync_protocol")),
+        scheduler_{std::move(scheduler)},
```

Comment:
same. Ensure not nullptr

Reply:
btw, I don't see that scheduler_ is used anyhow

Reply:
`scheduler_` is used in `RequestResponseTimeout::wrap`